v3/src/components/axis/helper-models/axis-helper.ts

```diff
@@ -0,0 +1,76 @@
+import { select } from "d3"
+import { isVertical } from "../../axis-graph-shared"
+import { AxisBounds, AxisPlace, axisPlaceToAxisFn } from "../axis-types"
+import { IAxisModel } from "../models/axis-model"
+import { IDataDisplayContentModel } from "../../data-display/models/data-display-content-model"
+import { IAxisLayout } from "../models/axis-layout-context"
+import { MultiScale } from "../models/multi-scale"
+
+export class AxisHelper {
+  multiScale: MultiScale | undefined
+
+  constructor(
+    protected displayModel: IDataDisplayContentModel,
+    protected subAxisIndex: number,
+    protected axisPlace: AxisPlace,
+    protected subAxisElt: SVGGElement | null,
+    protected axisModel: IAxisModel,
+    protected layout: IAxisLayout,
+    protected isAnimating: () => boolean
```

Comment:
Given how complicated the derived class constructors become, I'm inclined to think that the constructor shorthand syntax is not worth it in this case. I'd recommend just going back to taking an args object like:
```typescript
export interface IAxisHelperArgs {
  displayModel: IDataDisplayContentModel
  subAxisIndex: number
  subAxisElt: SVGGElement | null
  axisModel: IAxisModel
  layout: IAxisLayout
  isAnimating: () => boolean
}
```
and then the derived classes simplify to something like:
```typescript
interface IDateAxisHelperArgs extends IAxisHelperArgs {
  subAxisSelectionRef: MutableRefObject<Selection<SVGGElement, any, any, any> | undefined>
}

export class DateAxisHelper extends AxisHelper {
  subAxisSelectionRef: MutableRefObject<Selection<SVGGElement, any, any, any> | undefined>
  maxNumberExtent: number = kDefaultFontHeight

  constructor(props: IDateAxisHelperArgs) {
    super(props)
    this.subAxisSelectionRef = props.subAxisSelectionRef
  }
```
It's true that the redundancy that led us to consider the constructor shorthand syntax is still present, but in this case I think the redundancy is preferable to the alternative.